tutorials/terraform/data-proc-data-exchange-with-mch.tf

```
@@ -0,0 +1,282 @@
+  folder_id = local.folder_id
+}
+
+# Role to create Data Proc cluster
+resource "yandex_resourcemanager_folder_iam_binding" "dataproc" {
+  folder_id = data.yandex_resourcemanager_folder.my-folder.id
+  role      = "mdb.dataproc.agent"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.dataproc.id}",
+  ]
+}
+
+# Role to create S3 Bucket
+resource "yandex_resourcemanager_folder_iam_binding" "bucket-creator" {
+  folder_id = data.yandex_resourcemanager_folder.my-folder.id
+  role      = "editor"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.dataproc.id}",
+  ]
+}
+
+# Managed Service for ClickHouse cluster
+resource "yandex_mdb_clickhouse_cluster" "clickhouse-cluster" {
+  name               = "clickhouse-cluster"
+  environment        = "PRODUCTION"
+  description        = "Managed Service for ClickHouse cluster"
+  network_id         = local.network_id
+  security_group_ids = [yandex_vpc_security_group.clickhouse-and-vm-security-group.id]
+
+  clickhouse {
+    resources {
+      resource_preset_id = "s2.micro"
+      disk_type_id       = "network-ssd"
+      disk_size          = 10 # GB
+    }
+  }
+
+  host {
+    type             = "CLICKHOUSE"
+    zone             = "ru-central1-a"
+    subnet_id        = local.subnet_id
+    assign_public_ip = true # Required for connection from the Internet
+  }
+
+  database {
+    name = "db1"
+  }
+
+  user {
+    name     = "user1"
+    password = local.ch_password
+    permission {
+      database_name = "db1"
+    }
+  }
+}
+
+# VM in Yandex Compute Cloud
+resource "yandex_compute_instance" "vm-1" {
+  name        = "linux-vm"
+  platform_id = "standard-v3" # Intel Ice Lake
+  zone        = "ru-central1-a"
+
+  resources {
+    cores  = 2
+    memory = 4 # GB
+  }
+
+  boot_disk {
+    initialize_params {
+      image_id = "fd8ciuqfa001h8s9sa7i" # Ubuntu 20.04
+    }
+  }
+
+  network_interface {
+    subnet_id          = local.subnet_id
+    security_group_ids = [yandex_vpc_security_group.clickhouse-and-vm-security-group.id]
+    nat                = true # Required for connection from the Internet
+  }
+
+  metadata = {
+    ssh-keys = local.vm_ssh_keys
+  }
+}
+
+# S3 Bucket
+resource "yandex_iam_service_account_static_access_key" "my-bucket-key" {
+  service_account_id = yandex_iam_service_account.dataproc.id
+}
+
+resource "yandex_storage_bucket" "my-bucket" {
+  depends_on = [
+    yandex_resourcemanager_folder_iam_binding.bucket-creator
+  ]
+
+  bucket     = "my-bucket" # Should be unique in Cloud
+  access_key = yandex_iam_service_account_static_access_key.my-bucket-key.access_key
+  secret_key = yandex_iam_service_account_static_access_key.my-bucket-key.secret_key
+}
+
+# Data Proc cluster
+resource "yandex_dataproc_cluster" "my-dp-cluster" {
+  depends_on  = [yandex_resourcemanager_folder_iam_binding.dataproc]
+  bucket      = yandex_storage_bucket.my-bucket.bucket
+  description = "Dataproc Cluster created by Terraform"
```

По-моему, вот этого достаточно:

```suggestion
  description = "DataProc cluster"
```